src/main/java/com/example/springboot/dao/DAOInterface.java

```diff
@@ -0,0 +1,42 @@
+package com.example.springboot.dao;
+
+import java.util.Collection;
+
+/**
+ * This interface defines a methods for work with database
+ * @author Dmytro Kohut
+ * @version 1.1
+ */
+public interface DAOInterface<T> {
```

Comment:
`DAOInterface` really? WTF? You have to create new interface for each DAO, not one abstract for all of them